### PR TITLE
fix: Do not use aspect ratio of 1 for residual plot

### LINKED ISSUE #1292: Estimator Report in regression: plot doesn't display correctly
### Describe the bug

The prediction_error plot doesn't display correctly in some cases.  

### Steps/Code to Reproduce

```python
from sklearn.datasets import make_regression
from sklearn.linear_model import LinearRegression
from skore import EstimatorReport, ComparisonReport

X, y = make_regression(n_targets=2)
estimator = LinearRegression()
report_reg = EstimatorReport(estimator, X_train=X, y_train=y, X_test=X, y_test=y)

comp_reg = ComparisonReport(reports=[report_reg, report_reg], n_jobs = -1)
comp_reg.metrics.plot.prediction_error()
```

### Expected Behavior

- we should see the whole axis
- as much as possible, the legend shouldn't overlap the axis

### Actual Behavior

Image

### Environment

```shell
System:
    python: 3.12.8 | packaged by Anaconda, Inc. | (main, Dec 11 2024, 16:31:09) [GCC 11.2.0]
executable: /home/marie/anaconda3/envs/skore_test/bin/python
   machine: Linux-6.8.0-52-generic-x86_64-with-glibc2.35

Python dependencies:
        skore: 0.0.0+unknown
          pip: 24.2
   setuptools: 75.1.0
    diskcache: 5.6.3
      fastapi: 0.115.6
       joblib: 1.4.2
   matplotlib: 3.9.3
        numpy: 2.2.0
       orjson: 3.10.15
       pandas: 2.2.3
 platformdirs: 4.3.6
       plotly: 5.24.1
       psutil: 5.9.0
      pyarrow: 18.1.0
         rich: 13.9.4
 scikit-learn: 1.6.0
        skops: 0.11.0
      uvicorn: 0.32.1

Current version of the branch comparator (as of Feb 6th morning), this piece of skore code corresponds to skore 0.6.1.
```

### PR DESCRIPTION
**Pull request by glemaitre**
closes #1292 

We forced the aspect ratio of 1.0 for the residual plot. However the residual is computing `actual vs predicted` and the scale of the residual can be really different from the prediction if the error is low, explaining the problem of the plot.

This PR only set the aspect ratio for the actual vs predicted plot only.

### CODE REVIEW

**Review by auguste-probabl — APPROVED** [review]
Great, thanks!
